ls1intum/Artemis
# Show Build success if no tests are executed

### Checklist

### Motivation and Context


During an exam some courses don't provide any tests for programming exercises. But the compilation is still tested.
The feedback "0 of 0 tests passed" is not a helpful message in an exam. This should have a better message.

### Description

If no tests are present, but the build succeeded, a new message is shown in programming exercises as a score, that shows that the build compiled successfully.

### Steps for Testing


1. New Feature
    1. Create a programming exercise that doesn't have any tests that run before the due date. (No tests at all lead to a build failure)
    2. Submit a compiling code as student
    3. See the new compilation message.
2. Old exercises: make sure that normal exercises with tests still work:
    - A submission with no tests passing, but some tests present
    - A submission with build failure
    - A submission with some and one with all passing tests
    - A submission with manual feedback should work as normal

### Screenshots


![Bildschirmfoto 2021-04-27 um 16 40 54](https://user-images.githubusercontent.com/38322605/116260856-5db4aa00-a777-11eb-8962-e0897af6cfba.png)
